src/plugins/console/public/services/autocomplete.ts

```diff
@@ -28,53 +54,165 @@ export class AutocompleteInfo {
+        // a scheduled request if the user turns off polling.
+        if (settings.getPolling()) {
+          this.retrieve(settings, settings.getAutocomplete());
+        }
+      }, settings.getPollInterval());
+    });
+  }
+
+  private retrieveSettings<T>(
+    settingsKey: SettingsKey,
+    settingsToRetrieve: SettingsToRetrieve
+  ): Promise<T | {}> {
+    const settingKeyToPathMap = {
+      fields: '_mapping',
+      indices: '_aliases',
+      legacyTemplates: '_template',
+      indexTemplates: '_index_template',
+      componentTemplates: '_component_template',
+      dataStreams: '_data_stream',
+    };
+
+    // Fetch autocomplete info if setting is enabled and if user has made changes.
+    if (settingsToRetrieve[settingsKey]) {
+      return send({
+        http: this.http,
+        method: 'GET',
+        path: `${settingKeyToPathMap[settingsKey]}?pretty=false`, // pretty=false to compress the response and save bandwidth by avoiding whitespace
+        asSystemRequest: true,
       });
+    } else {
+      // If the user doesn't want autocomplete suggestions, then clear any that exist
+      return Promise.resolve({});
+    }
+  }
+
+  private async retrieveMappings(settingsToRetrieve: SettingsToRetrieve) {
+    const mappings = await this.retrieveSettings<IndicesGetMappingResponse>(
+      ENTITIES.FIELDS,
+      settingsToRetrieve
+    );
+
+    if (mappings) {
+      const maxMappingSize = Object.keys(mappings).length > 10 * 1024 * 1024;
+      let mappingsResponse;
+
+      if (maxMappingSize) {
+        this.notifications.toasts.addWarning({
+          title: i18n.translate('console.autocomplete.mappingsTooLargeTitle', {
+            defaultMessage: 'Mappings too large',
+          }),
+          text: i18n.translate('console.autocomplete.mappingsTooLargeText', {
+            defaultMessage: 'Mappings are too large to be displayed in autocomplete.',
+          }),
+        });
+        mappingsResponse = {};
+      } else {
+        mappingsResponse = mappings;
+      }
+      this.mapping.loadMappings(mappingsResponse);
+    }
+  }
+
+  private async retrieveAliases(settingsToRetrieve: SettingsToRetrieve) {
+    const aliases = await this.retrieveSettings<IndicesGetAliasResponse>(
+      ENTITIES.INDICES,
+      settingsToRetrieve
+    );
+
+    if (aliases) {
+      const collaborator = this.mapping;
+      this.alias.loadAliases(aliases, collaborator);
+    }
+  }
+
+  private async retrieveDataStreams(settingsToRetrieve: SettingsToRetrieve) {
+    const dataStreams = await this.retrieveSettings<IndicesGetDataStreamResponse>(
+      ENTITIES.DATA_STREAMS,
+      settingsToRetrieve
+    );
+
+    if (dataStreams && 'data_streams' in dataStreams) {
+      this.dataStream.loadDataStreams(dataStreams);
+    }
+  }
+
+  private async retrieveTemplates(settingsToRetrieve: SettingsToRetrieve) {
+    const legacyTemplates = await this.retrieveSettings<IndicesGetTemplateResponse>(
```

Comment:
is it possible to run legacyTemplates, indexTemplates, componentTemplates in parallel?